File: mobile/src/entities/family/lib/carePushCopy.ts
import type { CarePushKind } from '@/entities/family/api/invoke-care-push';
import { COPY } from '@/shared/copy';

/** 안심 푸시 카피 — 점수·감시 톤 금지 */
export function buildCarePushCopy(
  kind: CarePushKind,
  actorNickname?: string | null,
): { title: string; body: string } {
  const who = actorNickname?.trim() || COPY.family.memberFallback;
  if (kind === 'taken') {
    return {
      title: COPY.push.careTakenTitle,
      body: COPY.push.careTakenBody(who),
    };
  }
  return {
    title: COPY.push.careStuckTitle,
    body: COPY.push.careStuckBody(who),
  };
}

File: android-app/core/src/main/kotlin/com/jinlabs/yakok/core/family/CarePushCopy.kt
package com.jinlabs.yakok.core.family

import com.jinlabs.yakok.core.constants.CarePush
import com.jinlabs.yakok.core.copy.Copy

enum class CarePushKind {
    Taken,
    StuckEscalate,
    ;

    val wire: String
        get() = when (this) {
            Taken -> "taken"
            StuckEscalate -> "stuck_escalate"
        }

    val channelId: String
        get() = when (this) {
            Taken -> CarePush.ChannelTaken
            StuckEscalate -> CarePush.ChannelStuck
        }
}

data class CarePushMessage(
    val title: String,
    val body: String,
)

object CarePushCopy {
    fun build(kind: CarePushKind, actorNickname: String?): CarePushMessage {
        val who = actorNickname?.trim().orEmpty().ifEmpty { Copy.Family.MemberFallback }
        return when (kind) {
            CarePushKind.Taken -> CarePushMessage(
                title = Copy.Push.CareTakenTitle,
                body = Copy.Push.careTakenBody(who),
            )
            CarePushKind.StuckEscalate -> CarePushMessage(
                title = Copy.Push.CareStuckTitle,
                body = Copy.Push.careStuckBody(who),
            )
        }
    }
}
